test/PowerShellEditorServices.Test/Session/WorkspaceTests.cs

```diff
@@ -70,5 +70,23 @@ public void CanDetermineIsPathInMemory()
                     $"Testing path {testCase.Path}");
             }
         }
+
+        [Theory()]
+        [InlineData("file:///C:/banana/", @"C:\banana\")]
+        [InlineData("file:///C:/banana/ex.ps1", @"C:\banana\ex.ps1")]
+        [InlineData("file:///E:/Path/to/awful%23path", @"E:\Path\to\awful#path")]
+        [InlineData("file:///path/with/no/drive", @"C:\path\with\no\drive")]
+        [InlineData("file:///path/wi[th]/squ[are/brackets/", @"C:\path\wi[th]\squ[are\brackets\")]
+        [InlineData("file:///Carrots/A%5Ere/Good/", @"C:\Carrots\A^re\Good\")]
+        [InlineData("file:///Users/barnaby/%E8%84%9A%E6%9C%AC/Reduce-Directory", @"C:\Users\barnaby\脚本\Reduce-Directory")]
+        [InlineData("file:///C:/Program%20Files%20(x86)/PowerShell/6/pwsh.exe", @"C:\Program Files (x86)\PowerShell\6\pwsh.exe")]
```

Comment:
VSCode sends the drive letter escaped as well, so `C:\` would be `C%3A`.

Unfortunately, that doesn't work as well with `Uri`

```powershell
[Uri]::new("file:///C%3A/Program%20Files%20(x86)/PowerShell/6/pwsh.exe").LocalPath
# /C:/Program Files (x86)/PowerShell/6/pwsh.exe
```

Note the leading slash

Reply:
Oh good point I missed that one

Reply:
Ohhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh, so VSCode is sending us something incompatible with the .NET `Uri` class that has [been in there since Internet Explorer](https://blogs.msdn.microsoft.com/ie/2006/12/06/file-uris-in-windows/).

I think we will have to special case this, but I'm going to open an issue on VSCode to see what they say.